package/libgluonutil/src/libgluonutil.c

```diff
@@ -212,6 +214,40 @@ char * gluonutil_get_domain(void) {
 	return ret;
 }
 
+char * gluonutil_get_primary_domain(void) {
+	if (!gluonutil_has_domains())
+		return NULL;
+
+	char *domain_code = gluonutil_get_domain();
+	if (!domain_code)
+		return NULL;
+
+	const char *domain_path_fmt = "/lib/gluon/domains/%s.json";
+	char domain_path[strlen(domain_path_fmt) + strlen(domain_code)];
+	snprintf(domain_path, sizeof(domain_path), domain_path_fmt, domain_code);
+	free(domain_code);
+
+	char primary_domain_path[PATH_MAX+1] = "";
```

Comment:
Remove this initialization, only set the terminating 0 at the length returned by `readlink`.